src/main/java/Gertrude.java

```diff
@@ -0,0 +1,286 @@
+        String description = "";
+        String deadline = "";
+        boolean hasBy = false;
+        boolean isDeadline = false;
+
+        for(int i = 1; i < lineInputArr.length; i++) {
+            if (lineInputArr[i].equals("/by")) {
+                isDeadline = true;
+                hasBy = true;
+            } else if (!isDeadline) {
+                description = description + lineInputArr[i] + " ";
+            } else {
+                deadline = deadline + lineInputArr[i] + " ";
+            }
+        }
+
+        if(description.isEmpty()) {
+            System.out.println("Description is missing");
+            throw new GertrudeException();
+        }
+        if(!hasBy) {
+            System.out.println("/by is missing.");
+            throw new KeywordException();
+        }
+        if(deadline.isEmpty()) {
+            System.out.println("Deadline is empty.");
+            throw new GertrudeException();
+        }
+
+        Deadline newDeadline = new Deadline(description, deadline);
+        addTask(newDeadline, description, tasks);
+    }
+
+    public static void addEvent(String[] lineInputArr, ArrayList<Task> tasks) throws GertrudeException, KeywordException {
+        String description = "";
+        String start = "";
+        String end = "";
+        String section = "description";
+        boolean hasFrom = false;
+        boolean hasTo = false;
+        for(int i = 1; i < lineInputArr.length; i++) {
+            if (lineInputArr[i].equals("/from")) {
+                section = "from";
+                hasFrom = true;
+            } else if (lineInputArr[i].equals("/to")) {
+                section = "to";
+                hasTo = true;
+            } else if (section.equals("description")) {
+                description += lineInputArr[i] + " ";
+            } else if (section.equals("from")) {
+                start += lineInputArr[i] + " ";
+            } else if (section.equals("to")) {
+                end += lineInputArr[i] + " ";
+            }
+        }
+
+        if(description.isEmpty()) {
+            System.out.println("Description is empty.");
+            throw new GertrudeException();
+        }
+        if(!hasFrom) {
+            System.out.println("/from is missing.");
+            throw new KeywordException();
+        }
+        if(start.isEmpty()) {
+            System.out.println("From is empty.");
+            throw new GertrudeException();
+        }
+        if(!hasTo) {
+            System.out.println("/to is missing.");
+            throw new KeywordException();
+        }
+        if(end.isEmpty()) {
+            System.out.println("To is empty.");
+            throw new GertrudeException();
+        }
+
+        Event newEvent = new Event(description, start, end);
+        addTask(newEvent, description, tasks);
+    }
+
+    public static ArrayList<Task> openFile() throws IOException, CorruptedFileException, IndexOutOfBoundsException {
```

Comment:
You could have a FileManager class that handles the file reading and writing independently of the logic in Gertrude. This improves code reusability and reduces the responsibilities of each class.